src/containerapp/azext_containerapp/tests/latest/custom_preparers.py

```diff
@@ -56,3 +56,49 @@ def _get_resource_group(self, **kwargs):
                        'decorator @{} in front of this preparer.'
             raise CliTestError(template.format(ResourceGroupPreparer.__name__,
                                                self.resource_group_parameter_name))
+
+
+class SubnetPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
+    def __init__(self, name_prefix='vnet', location='eastus2euap', resource_group_parameter_name='resource_group', vnet_name=None, vnet_address_prefixes='14.0.0.0/23', subnet_address_prefixes='14.0.0.0/23',
+                 delegations='Microsoft.App/environments', subnet_name="sub", service_endpoints=None, skip_delete=False):
+        super(SubnetPreparer, self).__init__(name_prefix, 15)
+        self.cli_ctx = get_dummy_cli()
+        self.location = location
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.vnet_name = vnet_name
+        if vnet_name is None:
+            self.vnet_name = self.create_random_name()
+        self.vnet_address_prefixes = vnet_address_prefixes
+        self.subnet_address_prefixes = subnet_address_prefixes
+        self.delegations = delegations
+        self.subnet_name = subnet_name
+        self.service_endpoints = service_endpoints
+        self.skip_delete = skip_delete
+
+    def create_resource(self, name, **kwargs):
+        resource_group = self._get_resource_group(**kwargs)
+        subnet_id = None
+        try:
+            self.live_only_execute(self.cli_ctx, f"az network vnet create --address-prefixes {self.vnet_address_prefixes} -g {resource_group} -n {self.vnet_name} --location {self.location}")
+            subnet_command = f"az network vnet subnet create --address-prefixes {self.subnet_address_prefixes} " \
+                             f"--delegations {self.delegations} " \
```

Comment:
I think the delegation also should be optional. What about we need no delegation subs one day?